src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    public User login(String email, String password) {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        User user = userFromDbOptional.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(hashedPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Wrong email: "
+                + email + " or password for authenticate user: " + password);
+    }
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        user.setPassword(password);
+        try {
+            userService.add(user);
+            return user;
+        } catch (RuntimeException e) {
+            throw new RegistrationException("Can't register user with email: "
+                    + email + ", and password " + password, e);
+        }
+    }
```

Comment:
use checks like these instead of try catch
![image](https://github.com/mate-academy/hibernate-user-service-hw/assets/49767049/1248c864-e42a-4dd1-a701-f02148f22597)